src/main/java/uk/ac/ox/oucs/sirlouie/daia/Document.java

```diff
@@ -60,13 +60,48 @@ public void addItems(Collection<SearObject> beans) {
 				item.setHref(library.getURL());
 				item.setLabel(library.getLabel());
 				item.setStorage(library.getCollection());
+				item.setLibcode(library.getLibrary());
+				item.setLibname(library.getLibraryName());
+				item.setItemshelf(library.getCallNumber());
+				item.setItemdesc(library.getDescription());
+				item.setItemtype(library.getType());
+				if (item.getAvailability()!=null && item.getAvailability().equals("Available")){
+					item.setAvailableitems(1);
+				}
+				else {
+					item.setAvailableitems(0);
+				}
+				item.setTotalitems(1);
+				item.setAvailability(library.getAvailability());
+				item.setMapurl("");
 				Available service = new Available("loan");
 				service.setHref(library.getAvailableURL());
 				item.addAvailableService(service);
 				if (null != library.getLibrary()) {
 					item.setDepartment(new Department(library.getLibrary()));
 				}
-				items.add(item);
+
+				boolean alreadyExists = false;
+				for (Item existingItem : items) {
```

Comment:
If it's here I don't have an upper limit but from testing different books I've never seen more than 20-25 items.  I'm not sure I'll do it like this for the final release.